src/main/scala/components/memory/base-memory-components.scala

```diff
@@ -0,0 +1,43 @@
+// Base memory classes
+
+package dinocpu
+
+import chisel3._
+import chisel3.util.experimental.loadMemoryFromFile
+
+/**
+  * Base class for all modular backing memories. Simply declares the IO and the memory file.
+  */
+class BaseDualPortedMemory(size: Int, memfile: String) extends Module {
+  def wireMemPipe(portio: MemPortBusIO): Unit = {
```

Comment:
This function is only used in the `DualPortedCombinMemory`, right? The `NonCombin` version has a different version (with a different signature) of this function. I think it would be better to move this into `DualPortedCombinMemory`